Update framework

peppy opened the pull request:
- [x] Framework bump
- [x] Test fixes for pooling changes

peppy committed:
Update IPC usages

Of note, I've disabled IPC on visual test runners as we generally don't
use IPC in these cases. Having it set means that the game will not open
while visual tests are open, which has been a complaint from devs in the
past.

frenzibyte commented on osu.Desktop/Program.cs:
```diff
@@ -102,7 +102,7 @@ public static void Main(string[] args)
             }
 
-            using (DesktopGameHost host = Host.GetSuitableDesktopHost(gameName, new HostOptions { BindIPC = !tournamentClient }))
+            using (DesktopGameHost host = Host.GetSuitableDesktopHost(gameName, new HostOptions { IPCPort = !tournamentClient ? OsuGame.IPC_PORT : null }))
```
Can we give debug builds a different IPC port so one can run a release version of lazer while a debug version is already running? That has been bothering me for quite some time now.

peppy replied:
I guess yeah

bdach replied:
> so one can run a release version of lazer while a debug version is already running

I don't ever recall needing to do that though? What's the rationale?

frenzibyte replied:
Often times I would be on a local version of lazer and then notice something weird. Fastest way for me to check it's not my fault or it's not a regression would be to fire up a release instance and check if it happens there.

peppy committed:
Allow debug instances to coexist alongside release instances

peppy committed:
Update framework and other nuget packages

- Moq held back because dicks
- NUnit held back because large API changes (trivial but effort)
- SignalR held back due to api deprecations

peppy committed:
Fix failing tests due to pooling safety changes

peppy committed:
Merge branch 'master' into update-framework
